index.js

```diff
@@ -39,16 +62,41 @@ const getAccounts = () => {
   })
 }
 
+const getSpecialTemplateParams = (command) => {
+  let returnType = ""
+  let returnExpression = ""
+
+  const pluck = (arr, key) => arr.map(item => item[key])
+
+  returnType = pluck(specialGlobals[command], "returnType").join(", ")
+  returnExpression = pluck(specialGlobals[command], "prop")
+  returnExpression = "(" + returnExpression.map(item => command + "." + item).join(", ") + ");"
+
+  return {"returnType": returnType, "returnExpression": returnExpression}
+}
+
 /** Takes a list of commands and evaluates them inside a contract. Returns a promised result of the last command. Returns null if the command is not an expression. */
 const evalSol = (commands, options={}) => {
   /** Returns true if the given command is an expression that can return a value. */
   const isExpression = command => (!/[^=]=[^=]/.test(command)) && (!command.startsWith('delete'))
   const lastCommand = commands[commands.length - 1]
+  const commandWithoutSemi = lastCommand.endsWith(';') ? lastCommand.substring(0, lastCommand.length - 1) : lastCommand
 
   let returnType = 'bool'
   const content = commands.slice(0, commands.length - 1).join('\n')
-  const contentWithReturnExpression = commands.join('\n')
-  const sourceFirstPass = template({ content: contentWithReturnExpression, returnType, returnExpression: isExpression(lastCommand) ? lastCommand : 'false;' })
+  let contentWithReturnExpression = commands.join('\n')
+
+  //let sourceFirstPass = template({ content: contentWithReturnExpression, returnType, returnExpression: isExpression(lastCommand) ? lastCommand : 'false;' })
+  const specialCommand = specialGlobals[commandWithoutSemi]
+  const sourceFirstPass = template(specialCommand ? {
+    content,
+    returnType: getSpecialTemplateParams(commandWithoutSemi).returnType,
+    returnExpression: getSpecialTemplateParams(commandWithoutSemi).returnExpression
```

Comment:
Here instead of calling `getSpecialTemplateParams`, just inline the `pluck` + `join`.